core/src/edu/asu/c3simulator/widgets/NavigationPanel.java

```diff
@@ -113,28 +146,12 @@ public void addButton(String buttonName, Screen targetedScreen)
 	 *            The screen that the button maps to. If null, no actionListener will be
 	 *            added to the button.
 	 */
-	public boolean addSubButton(String parentButtonName, String buttonName,
+	public void addSubButton(String parentButtonName, String buttonName,
 			Screen targetedScreen)
 	{
-		String storedParentButtonName = storedButtonName(parentButtonName);
-		if (!storedParentButtonName.equalsIgnoreCase(""))
-		{
-			if (subButtons.containsKey(storedParentButtonName))
-			{
-				LinkedHashMap<String, Screen> relativeSubButtons = subButtons
-						.get(storedParentButtonName);
-				relativeSubButtons.put(buttonName, targetedScreen);
-			}
-			else
-			{
-				LinkedHashMap<String, Screen> newSubButtons = new LinkedHashMap<String, Screen>();
-				newSubButtons.put(buttonName, targetedScreen);
-				subButtons.put(storedParentButtonName, newSubButtons);
-			}
-			recreateNavigationPanel();
-			return true;
-		}
-		return false;
+		ButtonNodeListener parent = buttons.get(parentButtonName);
```

Comment:
Question. If the user of this class doesn't use the same field for the parentButtonName and instead create a new string everytime, would that affect the how this method functions?
Ex. [
-------instead of this-------
String parent = "SomeParentButton";
String sub= "SubButton";
addButton(parent , null);
addSubButton(parent, sub, null);
-------They do this-------
addButton("SomeParentButton", null);
addSubButton("SomeParentButton", "SubButton", null);
]